### PR TITLE
Properly format code in arrow project

### PR DESCRIPTION
**Pull request by nastra**
These are all purely whitespace changes that properly format the code (according to `intellij-java-palantir-style.xml`) in the `arrow` project. I'm planning to work on some stuff in that project, but don't want to "pollute" future PRs with formatting changes.

### CODE REVIEW

File: arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ColumnarBatch.java [hunk 1]
```diff
@@ -36,17 +36,17 @@
   ColumnarBatch(int numRows, ColumnVector[] columns) {
     for (int i = 0; i < columns.length; i++) {
       int columnValueCount = columns[i].getFieldVector().getValueCount();
-      Preconditions.checkArgument(numRows == columnValueCount,
+      Preconditions.checkArgument(
+          numRows == columnValueCount,
```
Comment (rdblue) [inline arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ColumnarBatch.java:40]: I don't have a problem with including the condition on the previous line. Is the rule that if one argument starts with a newline that all arguments must be?

**Comment by rdblue** [comment]
@nastra is your intent to enforce these style rules in CI as well? I think we want to consider whether it is worth making these changes because these will cause commit conflicts when cherry-picking commits before or after this is done. One reason to go ahead and do that is if we intend to enforce the style in CI and conform to it in all cases. I think that is also something we should do if we intend to make these changes. Better style enforcement seems like a good thing in general, as long as we aren't just periodically "fixing" without enforcement.

FYI @cwsteinbach.

File: arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ColumnarBatch.java (hunk 1 above)
Comment (nastra) [reply arrow/src/main/java/org/apache/iceberg/arrow/vectorized/ColumnarBatch.java:40]: as mentioned in the PR description I was just applying whatever formatting rules are defined in `intellij-java-palantir-style.xml`.

**Comment by nastra** [comment]
Personally I would prefer enforcing style rules, otherwise you'll always have the problem that the code in the repo isn't formatted as defined by the formatting rules and people trying to contribute will format their cause (using keyboard shortcuts due to muscle memory as in my case) and cause more whitespace changes than they intend to do. 

The problem however with e.g. `checkstyle:check` is that it tells you when the format isn't what it expects it to be, but you still have to apply formatting manually.
 
On the nessie project we're usign [spotless](https://github.com/diffplug/spotless) which allows to `check` but also has `apply`, which just auto-formats the code (see also https://github.com/diffplug/spotless/tree/main/plugin-maven) according to the style guide.

One of the problems that we were facing is that you'd like to have Eclipse / IntelliJ / cmd line apply the exact same code style and that seems to be the easiest with google-java-format which isn't everybody's favorite, but produces consistent formatting results.

Personally I stopped caring about the differences of different style rules. All I care about is that those can be enforced and easily applied via cmd line + IDE. Other programming languages (Go for example) solved this in the language and saved a huge amount of discussion hours among devs :)

If there's an interest in using `spotless` in Iceberg, then I would be willing to provide a PR.

**Pull request closed without merging**